modules/5_segmentation/src/segmentation.cpp

```diff
@@ -16,8 +17,68 @@ ADAS::ADAS() {
     // Load deep learning network into memory
     auto net = ie.ReadNetwork(join(DATA_FOLDER, "semantic-segmentation-adas-0001.xml"),
                               join(DATA_FOLDER, "semantic-segmentation-adas-0001.bin"));
+	// Specify preprocessing procedures
+	// (NOTE: this part is different for different models!)
+	InputInfo::Ptr inputInfo = net.getInputsInfo()["data"];
+	inputInfo->getPreProcess().setResizeAlgorithm(ResizeAlgorithm::RESIZE_BILINEAR);
+	inputInfo->setLayout(Layout::NHWC);
+	inputInfo->setPrecision(Precision::U8);
+	outputName = net.getOutputsInfo().begin()->first;
+
+	// Initialize runnable object on CPU device
+	ExecutableNetwork execNet = ie.LoadNetwork(net, "CPU");
+
+	// Create a single processing thread
+	req = execNet.CreateInferRequest();
+}
+
+static Blob::Ptr wrapMatToBlob(const Mat& m) {
+	//CV_Assert(m.depth() == CV_8U);
+	std::vector<size_t> dims = { 1, (size_t)m.channels(), (size_t)m.rows, (size_t)m.cols };
+	return make_shared_blob<uint8_t>(TensorDesc(Precision::U8, dims, Layout::NHWC),
+		(uint8_t*)m.data);
 }
 
 void ADAS::segment(const Mat& image, Mat& mask) {
-    CV_Error(Error::StsNotImplemented, "ADAS semantic segmentation");
+	// Create 4D blob from BGR image
+	Mat resized;
+	resize(image, resized, Size(1024, 2048));
```

Comment:
Why it's used? Moreover `Size(1024, 2048)` is 2048 height and 1024 width

Reply:
Because the input should be this size, as it's said here: https://github.com/opencv/open_model_zoo/blob/master/models/intel/semantic-segmentation-adas-0001/description/semantic-segmentation-adas-0001.md
I changed the size as Size(2048, 1024), but it still doesn't work.

Reply:
Yes, but at the same time we have automatic resize `inputInfo->getPreProcess().setResizeAlgorithm(ResizeAlgorithm::RESIZE_BILINEAR);`. Anyway, it's OK